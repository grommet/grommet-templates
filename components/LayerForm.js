'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Layer = require('grommet/components/Layer');

var _Layer2 = _interopRequireDefault(_Layer);

var _Box = require('grommet/components/Box');

var _Box2 = _interopRequireDefault(_Box);

var _Form = require('grommet/components/Form');

var _Form2 = _interopRequireDefault(_Form);

var _FormFields = require('grommet/components/FormFields');

var _FormFields2 = _interopRequireDefault(_FormFields);

var _Footer = require('grommet/components/Footer');

var _Footer2 = _interopRequireDefault(_Footer);

var _Button = require('grommet/components/Button');

var _Button2 = _interopRequireDefault(_Button);

var _Spinning = require('grommet/components/icons/Spinning');

var _Spinning2 = _interopRequireDefault(_Spinning);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } // (C) Copyright 2016 Hewlett Packard Enterprise Development LP

var LayerForm = function (_Component) {
  _inherits(LayerForm, _Component);

  function LayerForm() {
    _classCallCheck(this, LayerForm);

    var _this = _possibleConstructorReturn(this, (LayerForm.__proto__ || Object.getPrototypeOf(LayerForm)).call(this));

    _this._onSubmit = _this._onSubmit.bind(_this);
    return _this;
  }

  _createClass(LayerForm, [{
    key: '_onSubmit',
    value: function _onSubmit(event) {
      event.preventDefault();
      this.props.onSubmit();
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props;
      var submitLabel = _props.submitLabel;
      var onClose = _props.onClose;
      var title = _props.title;
      var compact = _props.compact;
      var busy = _props.busy;
      var secondaryControl = _props.secondaryControl;

      var control = void 0;
      if (busy) {
        var label = true === busy ? '' : busy;
        control = _react2.default.createElement(
          _Box2.default,
          { direction: 'row', align: 'center',
            pad: { horizontal: 'medium', between: 'small' } },
          _react2.default.createElement(_Spinning2.default, null),
          _react2.default.createElement(
            'span',
            { className: 'secondary' },
            label
          )
        );
      } else {
        control = _react2.default.createElement(_Button2.default, { type: 'submit', primary: true, label: submitLabel,
          onClick: this._onSubmit });
      }

      return _react2.default.createElement(
        _Layer2.default,
        { align: 'right', closer: true, onClose: onClose,
          a11yTitle: title },
        _react2.default.createElement(
          _Form2.default,
          { onSubmit: this._onSubmit, compact: compact },
          _react2.default.createElement(
            'h1',
            null,
            title
          ),
          _react2.default.createElement(
            _FormFields2.default,
            null,
            this.props.children
          ),
          _react2.default.createElement(
            _Footer2.default,
            { pad: { vertical: 'medium' }, justify: 'between' },
            control,
            secondaryControl
          )
        )
      );
    }
  }]);

  return LayerForm;
}(_react.Component);

exports.default = LayerForm;
;

LayerForm.propTypes = {
  busy: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.bool]),
  compact: _react.PropTypes.bool,
  onClose: _react.PropTypes.func.isRequired,
  onSubmit: _react.PropTypes.func.isRequired,
  secondaryControl: _react.PropTypes.node,
  submitLabel: _react.PropTypes.string.isRequired,
  title: _react.PropTypes.string.isRequired
};
module.exports = exports['default'];