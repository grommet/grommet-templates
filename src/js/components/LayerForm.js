// (C) Copyright 2016 Hewlett Packard Enterprise Development LP

import React, { PropTypes } from 'react';
import Layer from 'grommet/components/Layer';
import Box from 'grommet/components/Box';
import Form from 'grommet/components/Form';
import FormFields from 'grommet/components/FormFields';
import Footer from 'grommet/components/Footer';
import Button from 'grommet/components/Button';
import BusyIcon from 'grommet/components/icons/Spinning';

const LayerForm = (props) => {

  let control;
  if (props.busy) {
    const label = (true === props.busy ? '' : props.busy);
    control = (
      <Box direction="row" align="center"
        pad={{horizontal: 'medium', between: 'small'}}>
        <BusyIcon /><span className="secondary">{label}</span>
      </Box>
    );
  } else {
    control = (
      <Button type="submit" primary={true} strong={true}
        label={props.submitLabel}
        onClick={props.onSubmit} />
    );
  }

  return (
    <Layer align="right" closer={true} onClose={props.onClose}
      a11yTitle={props.title}>
      <Box full="vertical" justify="center">
        <Form onSubmit={props.onSubmit} compact={props.compact}>
          <h1>{props.title}</h1>
          <FormFields>
            {props.children}
          </FormFields>
          <Footer pad={{vertical: 'medium'}} justify="between">
            {control}
            {props.secondaryControl}
          </Footer>
        </Form>
      </Box>
    </Layer>
  );
};

LayerForm.propTypes = {
  busy: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  compact: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  secondaryControl: PropTypes.node,
  submitLabel: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

export default LayerForm;
