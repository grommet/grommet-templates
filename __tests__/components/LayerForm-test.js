// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import LayerForm from '../../src/js/components/LayerForm';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

global.fetch = () => Promise.resolve();
describe('LayerForm', () => {
  it('default', () => {
    const component = renderer.create(
      <LayerForm onClose={() => {}} onSubmit={()=> {}} title='Test'
        submitLabel='Submit' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
