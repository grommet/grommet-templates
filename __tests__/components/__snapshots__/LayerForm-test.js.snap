exports[`LayerForm default 1`] = `
<span
  style={
    Object {
      "display": "none"
    }
  } />
`;
